varname/core.py

```diff
@@ -287,17 +289,28 @@ def nameof(var, *more_vars, # pylint: disable=unused-argument
 
 def argname(arg: Any, # pylint: disable=unused-argument
             *more_args: Any,
+            # *, keyword-only argument, only available with python3.8+
+            func: Optional[Callable] = None,
             vars_only: bool = True) -> Union[str, Tuple[str]]:
-    """Get the argument names/sources of a function call
-
-    Note:
-        When vars_only is False, then asttokens has to be installed.
+    """Get the argument names/sources passed to a function
 
     Args:
         arg: Parameter of the function, used to map the argument passed to
             the function
-        *args: Other parameters of the function, used to map more arguments
+        *more_args: Other parameters of the function, used to map more arguments
             passed to the function
+        func: The function. If not provided, the AST node of the function call
+            will be used to fetch the function:
+            - If a variable (ast.Name) used as function, the `node.id` will
+                be used to get the function from `locals()` or `globals()`.
+            - If variable (ast.Name), attributes (ast.Attribute),
+                subscripts (ast.Subscript), and combinations of those and
+                literals used as function, `pure_eval` will be used to evaluate
+                the node
+            - If `pure_eval` is not installed or failed to evaluate, `eval`
+                will be used. A warning will be shown since malicious code may
+                be evaluated in this case.
```

Comment:
In this case it's probably not malicious code, but it may still have unwanted side effects.